### PR TITLE
[WIP] add from training to serving document

### PR DESCRIPTION
**Pull request by MRXLT**
add doc TRAIN_TO_SERVICE.md

### CODE REVIEW

File: doc/TRAIN_TO_SERVICE.md [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+            yield ("words", feas), ("label", label)
+
+        return data_iter
+```
+</details>
+
+映射之后的样本类似于以下的格式：
+
+```
+257 142 52 898 7 0 12899 1083 824 122 89527 134 6 65 47 48 904 89527 13 0 87 170 8 248 9 15 4 25 1365 4360 89527 702 89527 1 89527 240 3 28 89527 19 7 0 216 219 614 89527 0 84 89527 225 3 0 15 67 2356 89527 0 498 117 2 314 282 7 38 1097 89527 1 0 174 181 38 11 71 198 44 1 3110 89527 454 89527 34 37 89527 0 15 5912 80 2 9856 7748 89527 8 421 80 9 15 14 55 2218 12 4 45 6 58 25 89527 154 119 224 41 0 151 89527 871 89527 505 89527 501 89527 29 2 773 211 89527 54 307 90 0 893 89527 9 407 4 25 2 614 15 46 89527 89527 71 8 1356 35 89527 12 0 89527 89527 89 527 577 374 3 39091 22950 1 3771 48900 95 371 156 313 89527 37 154 296 4 25 2 217 169 3 2759 7 0 15 89527 0 714 580 11 2094 559 34 0 84 539 89527 1 0 330 355 3 0 15 15607 935 80 0 5369 3 0 622 89527 2 15 36 9 2291 2 7599 6968 2449 89527 1 454 37 256 2 211 113 0 480 218 1152 700 4 1684 1253 352 10 2449 89527 39 4 1819 129 1 316 462 29 0 12957 3 6 28 89527 13 0 457 8952 7 225 89527 8 2389 0 1514 89527 1
```
Comment (guru4elephant) [inline doc/TRAIN_TO_SERVICE.md:108]: 这里的格式对用户是不可见的，展示这个可能对理解训练帮助不大，是否删掉了也不影响用户理解？

File: doc/TRAIN_TO_SERVICE.md [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+            #在训练结束时使用PaddleServing中的模型保存接口保存出Serving所需的模型和配置文件
+            serving_io.save_model("{}_model".format(model_name),
+                                  "{}_client_conf".format(model_name),
+                                  {"words": data}, {"prediction": prediction},
+                                  fluid.default_main_program())
+```
+
+</details>
+
+执行loca_train.py脚本会进行训练并在训练结束时保存模型和配置文件。保存的文件分为imdb_cnn_client_conf和imdb_cnn_model文件夹，前者包含client端的配置文件，后者包含server端的配置文件和保存的模型文件。
```
Comment (guru4elephant) [inline doc/TRAIN_TO_SERVICE.md]: loca_train.py typo

Comment (guru4elephant) [reply doc/TRAIN_TO_SERVICE.md]: 可以展示一下tree的效果

File: doc/TRAIN_TO_SERVICE.md [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+    filelist = ["train_data/%s" % x for x in os.listdir("train_data")]
+    dataset.set_use_var([data, label])
+    pipe_command = "python imdb_reader.py"
+    dataset.set_pipe_command(pipe_command)
+    dataset.set_batch_size(128)
+    dataset.set_filelist(filelist)
+    dataset.set_thread(10)
+    #定义模型
+    avg_cost, acc, prediction = cnn_net(data, label, dict_dim)
+    optimizer = fluid.optimizer.SGD(learning_rate=0.01)
```
Comment (guru4elephant) [inline doc/TRAIN_TO_SERVICE.md]: CNN网络的训练一般用小batch会好一些，可以尝试batch_size = 4，learning_rate=0.001

File: doc/TRAIN_TO_SERVICE.md [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+
+以上命令中参数1为保存的server端模型和配置文件，参数2为工作目录会保存一些预测服务工作时的配置文件，该目录可以不存在但需要指定名称，预测服务会自行创建，参数3为端口号，参数4为词典文件。
+
+## Step9：明文数据调用预测服务
+启动完HTTP预测服务，即可通过一行命令进行预测：
+
+```
+curl -H "Content-Type:application/json" -X POST -d '{"words": "i am very sad | 0", "fetch":["prediction"]}' http://127.0.0.1:9292/imdb/prediction
+```
+预测流程正常时，会返回预测概率。
```
Comment (guru4elephant) [inline doc/TRAIN_TO_SERVICE.md]: 是否展示一下结果，并说明用户复现的结果可能和这个不同。

File: doc/TRAIN_TO_SERVICE.md (hunk 2 above)
Comment (MRXLT) [reply doc/TRAIN_TO_SERVICE.md]: 已修改，添加了模型训练的损失下降曲线

File: doc/TRAIN_TO_SERVICE.md (hunk 3 above)
Comment (MRXLT) [reply doc/TRAIN_TO_SERVICE.md]: 已修改，重新训练在65轮时测试集预测acc 88.19%

File: doc/TRAIN_TO_SERVICE.md (hunk 4 above)
Comment (MRXLT) [reply doc/TRAIN_TO_SERVICE.md]: 已添加

File: doc/TRAIN_TO_SERVICE.md (hunk 1 above)
Comment (MRXLT) [reply doc/TRAIN_TO_SERVICE.md:108]: 这里是为了解释为什么需要reader脚本、词典文件，这部分预处理在训练、部署服务中都有出现，感觉还是给个样例更能解释为什么会有预处理这一步。

File: doc/TRAIN_TO_SERVICE.md [hunk 5]
```diff
@@ -0,0 +1,362 @@
+# 使用PaddleServing快速搭建预测服务
```
Comment (guru4elephant) [inline doc/TRAIN_TO_SERVICE.md:1]: Paddle Serving

**Review by guru4elephant — APPROVED** [review]
LGTM
